api.swagger.yaml

```diff
@@ -467,96 +576,803 @@ paths:
+        '404':
+          description: The UUID does not correspond to an existing user.
+  '/users/{user_id}/groups':
+    get:
+      tags:
+        - users
+      description: List a user's groups.
+      parameters:
+        - in: path
+          name: user_id
+          description: The Invotra UUID of the user.
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: |
+            List of a user's groups.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/group-schema'
+        '400':
+          description: The supplied UUID was malformed.
+        '404':
+          description: The UUID does not correspond to an existing user.
+  '/users/{user_id}/follow':
+    delete:
+      tags:
+        - users
+      parameters:
+        - description: The Invotra UUID of the user to unfollow.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: Follower flag removed.
+        '400':
+          description: The UUID was malformed.
+        '404':
+          description: |
+            The user or follower flag does not exist.
+      description: Stop following a user.
+    get:
+      tags:
+        - users
+      parameters:
+        - description: The Invotra UUID of the user to check following status against.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: |
+            Follow status returned
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/common-status-schema'
+        '400':
+          description: The UUID was malformed.
+        '404':
+          description: The user does not exist.
+    put:
+      tags:
+        - users
+      parameters:
+        - description: The Invotra UUID of the user to follow.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The user is now following the other user.
+        '400':
+          description: The UUID was malformed.
+        '404':
+          description: The user or follower does not exist.
+      description: Follow a user as another user.
+  /groups:
+    get:
+      tags:
+        - groups
+      parameters:
```

Comment:
this either needs a pager, or external_id should be required

Reply:
added